src/parser/monk/brewmaster/modules/features/MitigationSheet.js

```diff
@@ -0,0 +1,460 @@
+import AgilityValue from './AgilityValue';
+import { diminish, ULDIR_K, MPLUS_K } from '../constants/Mitigation';
+
+
+function formatGain(gain) {
+  if(typeof gain === 'number') {
+    return formatNumber(gain);
+  } else if(gain.low !== undefined && gain.high !== undefined) {
+    return `${formatNumber(gain.low)} - ${formatNumber(gain.high)}`;
+  }
+  return null;
+}
+
+function formatWeight(gain, avg, norm, inc) {
+  if(typeof gain === 'number') {
+    return (gain / avg / norm * inc).toFixed(2);
+  } else if(gain.low !== undefined && gain.high !== undefined) {
+    return `${(gain.low / avg / norm * inc).toFixed(2)} - ${(gain.high / avg / norm * inc).toFixed(2)}`;
+  }
+  return null;
+}
+
+function calculateTotalGain(gain) {
+  const { low, high } = gain
+    .filter(({isLoaded}) => isLoaded !== false)
+    .reduce(({low, high}, {amount}) => {
+    if(typeof amount === 'number') {
+      low += amount;
+      high += amount;
+    } else if(amount.low !== undefined && amount.high !== undefined) {
+      low += amount.low;
+      high += amount.high;
+    }
+    return {
+      low, high,
+    };
+  }, { low: 0, high: 0 });
+
+  if(low === high) {
+    return low;
+  }
+
+  return { 
+    low, high,
+  };
+}
+
+function makeIcon(stat) {
+  const Icon = getIcon(stat);
+  return (
+    <Icon
+      style={{
+        height: '1.6em',
+        width: '1.6em',
+        marginRight: 10,
+      }}
+    />
+  );
+}
+
+function calculateLeatherArmorScaling(ilvl, amount, targetIlvl) {
+  return amount * Math.pow(1.0327, (targetIlvl - ilvl) / 5);
+}
+
+export default class MitigationSheet extends Analyzer {
+  static dependencies = {
+    masteryValue: MasteryValue,
+    agilityValue: AgilityValue,
+    cf: CelestialFortune,
+    stats: StatTracker,
+    stagger: Stagger,
+    gotox: GiftOfTheOx,
+  };
+
+  K = null;
+
+  armorDamageMitigated = 0;
+  versDamageMitigated = 0;
+  versHealing = 0;
+
+  static statsToAvg = ['agility', 'armor', 'versatility', 'mastery', 'crit'];
+  _lastStatUpdate = null;
+  _avgStats = {};
+
+  get masteryDamageMitigated() {
+    return this.masteryValue.expectedMitigation;
+  }
+
+  get masteryHealing() {
+    return this.masteryValue.totalMasteryHealing;
+  }
+
+  _critBonusHealing = 0;
+
+  get agiDamageMitigated() {
+    return this.agilityValue.totalAgiPurified;
+  }
+
+  get agiDamageDodged() {
+    return this.masteryValue.expectedMitigation - this.masteryValue.noAgiExpectedDamageMitigated;
+  }
+
+  get agiHealing() {
+    return this.agilityValue.totalAgiHealing;
+  }
+
+  get wdpsHealing() {
+    return this.gotox.wdpsBonusHealing;
+  }
+
+  get avgIlvl() {
+    const gear = this.selectedCombatant._combatantInfo.gear.filter(({id, itemLevel}) => id !== 0 && itemLevel > 1);
+    return gear.reduce((sum, {itemLevel}) => sum + itemLevel, 0) / gear.length;
+  }
+
+  constructor(...args) {
+    super(...args);
+
+    const fight = this.owner.fight;
+    if(fight.size === 5) {
+      this.K = MPLUS_K;
```

Comment:
I think this should be a getter. Also the name should be longer, how about `kFactor`? Just K is going to be hard to work with.